modules/core/src/lib/attribute-transition-utils.js

```diff
@@ -25,6 +26,54 @@ export function normalizeTransitionSettings(userSettings, layerSettings = {}) {
   return Object.assign({}, DEFAULT_TRANSITION_SETTINGS, layerSettings, userSettings);
 }
 
+// NOTE: NOT COPYING OVER OFFSET OR STRIDE HERE BECAUSE:
+// (1) WE DON'T SUPPORT INTERLEAVED BUFFERS FOR TRANSITIONS
+// (2) BUFFERS WITH OFFSETS ALWAYS CONTAIN VALUES OF THE SAME SIZE
+// (3) THE OPERATIONS IN THE SHADER ARE PER-COMPONENT (addition and scaling)
+export function getSourceBufferAttribute(gl, attribute) {
+  const {size, value, normalized, constant} = attribute;
+  // The Attribute we pass to Transform as a sourceBuffer must have {divisor: 0}
+  // so we create a copy of the attribute (with divisor=0) to use when running
+  // transform feedback
+  if (constant) {
+    // don't pass normalized here because the `value` from a normalized attribute is
+    // already normalized
+    return new BaseAttribute(gl, {constant, value, size});
+  }
+  const buffer = attribute.getBuffer();
+  return new BaseAttribute(gl, {divisor: 0, constant, buffer, size, normalized});
+}
+
+export function getAttributeTypeFromSize(size) {
+  switch (size) {
+    case 1:
+      return 'float';
+    case 2:
+      return 'vec2';
+    case 3:
+      return 'vec3';
+    case 4:
+      return 'vec4';
+    default:
+      throw new Error(`No defined attribute type for size "${size}"`);
+  }
+}
+
+export function cycleBuffers(buffers) {
+  const firstBuffer = buffers[0];
+  for (let i = 0; i < buffers.length - 1; i++) {
+    buffers[i] = buffers[i + 1];
+  }
+  buffers[buffers.length - 1] = firstBuffer;
+  return buffers;
```

Comment:
May be easier to just use `shift()` and `push()`?

Reply:
Ha! Good call. Much simpler :-)